galaxy_ng/app/api/v1/viewsets.py

```diff
@@ -0,0 +1,357 @@
+            assert kwargs.get('github_repo') is not None
+            assert kwargs.get('github_repo') != ''
+            if kwargs.get('alternate_role_name'):
+                assert kwargs.get('alternate_role_name') != ''
+        except Exception as e:
+            return e
+
+    def create(self, validated_data):
+
+        logger.info(f'github_user: {validated_data.data.get("github_user")}')
+        logger.info(f'github_repo: {validated_data.data.get("github_repo")}')
+        logger.info(f'github_reference: {validated_data.data.get("github_reference")}')
+        logger.info(f'alternate_role_name: {validated_data.data.get("alternate_role_name")}')
+
+        kwargs = {
+            'github_user': validated_data.data.get("github_user"),
+            'github_repo': validated_data.data.get("github_repo"),
+            'github_reference': validated_data.data.get("github_reference"),
+            'alternate_role_name': validated_data.data.get("alternate_role_name"),
+        }
+        error = self.validate_create_kwargs(kwargs)
+        if error:
+            return HttpResponse(str(error), status=403)
+
+        logger.info(f'SETTINGS: {settings}')
+        for x in dir(settings):
+            if 'auth' in x.lower():
+                logger.info(f'SETTING {x}: {getattr(settings, x)}')
+        logger.info(f'REQUEST HEADERS: {self.request.headers}')
+        logger.info(f'REQUEST USER: {self.request.user}')
+        logger.info(f'REQUEST USER IS AUTHENTICATED: {self.request.user.is_authenticated}')
+        logger.info(f'REQUEST USER USERNAME: {self.request.user.username}')
+
+        # validate is admin or gitub_user == auth_user ...
+        if not self.request.user.is_authenticated:
+            return HttpResponse('authentication required', status=403)
+        if not self.request.user.is_superuser and not \
+                self.request.user.username == kwargs['github_user']:
+            return HttpResponse('invalid permissions', status=403)
+
+        task = dispatch(legacy_role_import, kwargs=kwargs)
+        logger.info(f'TASK: {task}')
+
+        hashed = abs(hash(str(task.pulp_id)))
+        logger.info(f'NEW_TASK_ID: {task.pulp_id}')
+        logger.info(f'NEW_HASHED_TASK_ID: {hashed}')
+
+        role_name = kwargs['alternate_role_name'] or \
+            kwargs['github_repo'].replace('ansible-role-', '')
+
+        return Response({
+            'results': [{
+                'id': hashed,
+                'github_user': kwargs['github_user'],
+                'github_repo': kwargs['github_repo'],
+                'summary_fields': {
+                    'role': {
+                        'name': role_name
+                    }
+                }
+            }]
+        })
+
+    def get_task(self, request):
+        logger.info(f'GET TASK: {request}')
+        task_id = int(request.GET.get('id', None))
+        logger.info(f'GET TASK id: {task_id}')
+
+        this_task = None
+        for t in Task.objects.all():
```

Comment:
for when the number of tasks grows large, i wonder if doing an order by creation date could prevent iterating over a large number, since likely task lookups will be for recent tasks.

Reply:
done